docs/saved-filter-presets/database.md

```diff
@@ -0,0 +1,58 @@
+# Saved Asset Filter Presets – Database & Migration Spec
+
+## Objective
+Define the persistent data structures needed to store and manage saved filter presets for the advanced asset index while minimizing risk to existing data flows.
+
+## Prisma Schema Changes
+```prisma
+model AssetFilterPreset {
+  id              String   @id @default(cuid())
+  organizationId  String
+  ownerId         String
+  name            String
+  query           String
+  view            String
+  mode            AssetFilterPresetMode @default(ADVANCED)
+  isShared        Boolean  @default(false)
+  createdAt       DateTime @default(now())
+  updatedAt       DateTime @updatedAt
+  lastUsedAt      DateTime?
+
+  organization Organization @relation(fields: [organizationId], references: [id], onDelete: Cascade)
+  owner        User         @relation(fields: [ownerId], references: [id], onDelete: Cascade)
+
+  @@unique([organizationId, ownerId, name], map: "asset_filter_presets_owner_name_unique")
+  @@index([organizationId, isShared], map: "asset_filter_presets_org_shared_idx")
+  @@index([organizationId, lastUsedAt], map: "asset_filter_presets_last_used_idx")
+}
+
+enum AssetFilterPresetMode {
+  ADVANCED
+  SIMPLE
+}
+```
+
+### Notes
+- `mode` enum is future-proofing for simple index presets without requiring a breaking schema change.
+- `view` stores the current layout (`table`, `availability`, etc.).
+- `query` contains the sanitized query string (canonical `URLSearchParams` order) returned from `cleanParamsForCookie`.
+- `lastUsedAt` enables analytics-based ordering; nullable to avoid writes until first apply.
+
+## Migration Plan
+1. Generate migration script with `npm run db:prepare-migration -- saved-filter-presets`.
+2. Verify generated SQL adds enum before model to satisfy Postgres dependency order.
+3. Confirm indexes are created with descriptive names to avoid collisions.
```

Comment:
RLS enabling should be manually added to the end of the migration file. You can see it in previous migrations.